### ISSUE TITLE
[Feature Request] Option/Instruction to use Flatpak Gnome Boxes (useful for immutable OS)

### ISSUE THREAD

**Issue by bayazidbh** [casualsnek/cassowary#50]
Hi, I came from a mention on WinApps, which is neat, but also isn't very usable on immutable OS due to relying on qemu and virt-manager whereas only Gnome Boxes is officially available on Flathub right now. I've seen a project or two that tries to pack virt-manager as Flatpak using Flatpak Builder, but I'm not sure as to how well that works and how easy it'll be.

Since the SteamOS 3.0 is coming out soon, and that'll be on immutable OS, would it be possible to make an option to work with Gnome Boxes as well? An advantage would be that it should work with most other OS too because it is Flatpak-based. Thanks regardless.

**Comment by casualsnek** [comment]
Both of them can be used if IP address of windows guest is reachable from Linux host, the things that wont work is ip auto detection and auto suspend. I will check if anything can be done to make it work with gnome boxes !

**Comment by eylenburg** [comment]
Yes this would be great. And even better if Cassowary itself can be installed as a Flatpak which then connects to GNOME Boxes  (if that is possible)

**Comment by fenglengshun** [comment]
> And even better if Cassowary itself can be installed as a Flatpak which then connects to GNOME Boxes (if that is possible)

I don't think this is possible given that Flatpak is fairly against applications escaping from its sandbox and communicating with the rest of the system, nor does it make sense.

I think it'd make more sense as an AppImage or shipped like whatever Conty is, especially since in the case of Steam Deck you'll need to pre-pack the dependencies due to root overwriting by system updates is a bad user experience for this project.

**Comment by eylenburg** [comment]
> I think it'd make more sense as an AppImage or shipped like whatever Conty is, especially since in the case of Steam Deck you'll need to pre-pack the dependencies due to root overwriting by system updates is a bad user experience for this project.

You're right, AppImage is probably the best way for users on immutable systems
